classes/item.rb

```diff
@@ -0,0 +1,51 @@
+# 01 Create Item class in a separate .rb file.
+require 'date'
+
+class Item
+  def initialize(date_published, id)
+    @date_published = Date.parse(date_published)
+    @archived = false
+    @id = id
+  end
+
+  # 02
+  # Add all methods visible in the diagram.
+  def add_author(author)
+    @author = author
+    author.add_item(self)
+  end
+
+  def add_source(source)
+    @source = source
+    source.add_item(self)
+  end
+
+  def add_label(label)
+    @label = label
+    label.add_item(self)
+  end
+
+  def add_genre(genre)
+    @genre = genre
+    genre.add_item(self)
+  end
```

Comment:
- [ ] Please know that the Item class should have one Genre, Author, Label, and Source, to make sure that's the case you can do it like this:

```ruby
    def genre=(genre)
      @genre = genre
      genre.items.push(self) unless genre.items.include?(self)
    end

    def author=(author)
      @author = author
      author.items.push(self) unless author.items.include?(self)
    end

    def label=(label)
      @label = label
      label.items.push(self) unless label.items.include?(self)
    end
    
    def source=(source)
      @source = source
      source.items.push(self) unless source.items.include?(self)
    end
```